src/components/description_list/description_list.tsx

```diff
@@ -87,6 +40,16 @@ export const EuiDescriptionList: FunctionComponent<
   type = 'row',
   ...rest
 }) => {
+  const styles = euiDescriptionListStyles();
+  const isMobile = useIsWithinBreakpoints(['xs', 's']);
+
+  const cssStyles = [
+    styles.euiDescriptionList,
+    styles[align],
+    (type === 'column' || (type === 'responsiveColumn' && !isMobile)) &&
+      styles.column,
```

Comment:
Why use JS to detect screen size here? We should just the new `euiBreakpoint()` mixin to generate and handle responsive behavior via CSS media queries (which is going to be more performant than introducing a resize observer).

```suggestion
  const euiTheme = useEuiTheme();
  const styles = euiDescriptionListStyles(euiTheme);

  const cssStyles = [
    styles.euiDescriptionList,
    styles[type],
    styles[align],
```

Reply:
This was actually a change @cchaos made when she took a pass at this PR. I think the this was done for a future optimization that would allow consumers to adjust the breakpoints.

> PS: This also moved the logic of the breakpoint to the JS with useIsWithinBreakpoints(). This is nice because in the future we can allow consumer to the adjust that responsive breakpoint array.

https://github.com/elastic/eui/pull/5971#discussion_r907831171

Reply:
It's nice, but it's not performant (just my 2c). A resize observer is going to be more expensive than a CSS media query every time - especially since the current implementation is instantiating a resize observer _for every single DescriptionList component_ (if we absolutely *had* to use an observer, we'd want just 1 top level one probably instantiated in our context provider that returns a single `isMobile` flag).

However, there's still no need for a resize observer whatsoever - since we're using now Emotion, and one of the points of Emotion is to allow customizing CSS with JS, we can still allow customizing breakpoints in the future anyway by passing a `breakpoints` array prop to the `euiDescriptionListStyles()` function.

Reply:
Also, as an aside / out of curiosity, why did we even convert `euiBreakpoint()` as part of this PR if we weren't going to use it?

Reply:
Ok, no worries. I'll transfer this logic back to the styles file. 

> why did we even convert euiBreakpoint() as part of this PR if we weren't going to use it?
When I started the conversion, I converted the mixin and used it in `styles.ts`. This was before the suggestion from Caroline to move the logic to JS.

Reply:
Ahh gotcha, thanks for that clarification! Well the good news is I'd still really like to use `euiBreakpoint()` in this PR, so we'll def be using that work!

Also not sure if you've seen this if you're still working your way down my comments, but I have an example diff/commit to check out that drops the resize observer and uses `euiBreakpoint()` here: https://github.com/constancecchen/eui/commit/a9cccad387bfc4daf287f7ab72d4fc03e6880f8d